### PR TITLE
Completed Fixing Routes And Middlewares

### PR DESCRIPTION
**Pull request by key-joshua**
For this activity, you should update the route names in order to be more descriptive of the route functionanilty.

For example: in routes folder, a route file name should be named by <content>Route.ts → so it should be for stripe stripeRoute.ts.
inside this file you should handle all stripe routes so if main route for stripe is ‘/api-gateway/stripe/' then every HTTP method would be understanded this way:

router.get(<Route>) → router.get(’/products') should return a list of stripe products in this case.

router.post(<Route>) → router.post('/product') should create a new product for stripe.

router.put(<Route>) → router.put('/product') should update a product infromation for stripe.

router.delete(<Route>) → router.delete('/product/:id') should delete a stripe product.

This was an example of how routes should work and what it would be expected of each route, so for each other route corresponding to other services like users, business, etc should follow this structure so that is more comprehensive for developers to understand functionalities. 

Keep in mind that controllers attached to those routes should contain descriptive names and be in the controllers folder with it’s respective file name, for this case would be stripeController.ts

Middlewares should be defined by groups regarding each step of middleware interaction. For example if a middleware determinates user access to application ( maybe by JWT then the file containing this validations should be named authentication ( first step of access ). Then if a user that is already authenticated, needs a certain role or status to access a specific route then the file name should be authorization ( second step of access ). 

For our use cases this should be enough, but if needed I can send you a step verification guide to understand each step of access control.

### CODE REVIEW

File: src/routes/stripeRoute.ts [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+import { authorizationSchema } from '../modules/auth/validation/authValidation';
+import { isBodyValidation, isHeaderValidation, isQueryValidation } from '../middlewares/requestMiddleware';
+import { planDetailsSchema, checkoutSessionSchema, emailSchema } from '../modules/stripe/validation/stripeValidation';
+
+const router: Router = Router();
+
+router.get('/products', isHeaderValidation(authorizationSchema), gatewayAuthentication, stripeController.stripeGetProducts);
+router.post('/product', isHeaderValidation(authorizationSchema), gatewayAuthentication, isBodyValidation(planDetailsSchema), stripeController.stripecreateProduct);
+router.get('/subscriptions', isHeaderValidation(authorizationSchema), gatewayAuthentication, isQueryValidation(emailSchema), stripeController.stripeGetSubscription);
+router.post('/checkout-session', isHeaderValidation(authorizationSchema), gatewayAuthentication, isBodyValidation(checkoutSessionSchema), stripeController.stripeCheckoutSession);
```
Comment (Yopo-Nicolas) [inline src/routes/stripeRoute.ts]: isHeaderValidation and gatewayAuthentication are used in every route, so in this case, you should add this to the main route ('/stripe'). Same should be true for other routes that use same validations

**Review by Yopo-Nicolas — COMMENTED** [review]
Change this, other things seem okey

File: src/routes/stripeRoute.ts (hunk 1 above)
Comment (key-joshua) [reply src/routes/stripeRoute.ts]: Okay, noted

**Review by Yopo-Nicolas — APPROVED** [review]
Seems fine

**Pull request merged**